arduino_code/led_dimmer/led_dimmer.ino

```diff
@@ -110,10 +153,52 @@ int toggle_led() {
 }
 
 /* Turn led off, and store dim status */
-int turn_led_off() {
-	dimmer_target = 0;
+int all_led_off() {
+     
+    dimmer_speed = DIM_SPEED_FAST;
+    dimmer_target = 0;
+    
+}
+
+int all_led_on() {
+     
+    dimmer_speed = DIM_SPEED_FAST;
+    dimmer_target = 255;
+    
 }
 
+// set target and speed for wake up light
+int wake_set_target() {
+    
+    dimmer_speed = DIM_SPEED_SLOW;
+    dimmer_target = 255;
+    
+   
+    
+}
+
+
+int option_button_task(){
+    switch (board_id){
+            case(0 OR 1):
+            all_led_off();
+            Serial.println("LOCAL ALL LED OFF");
+            i2c_broadcast(I2C_ALL_OFF);
+            
+            case(2 OR 3):
+            all_led_on();
+            Serial.println("LOCAL ALL LED ON");
+            i2c_broadcast(I2C_ALL_ON);
+            
+            case(4):
+            wake_set_target();
+            Serial.println("LOCAL WAKE SET TARGET");
+            i2c_broadcast(I2C_WAKE_SET);
+    }
```

Comment:
- Je moet bovenaan je nieuwe I2C commandos wel nog defines
- defines of [magic constants](https://en.wikipedia.org/wiki/Magic_number_(programming)#Unnamed_numerical_constants) te vermijden.
- [juiste syntax voor een switch case statement](https://www.arduino.cc/reference/en/language/structure/control-structure/switchcase/)

Met de encoding die ik boven voorstelde zou dit trouwens simpeler zijn, aangezien enkel naar de waarde van de eerste twee jumpers moet kijken wat de actie is.

```
#define BOARD_ALL_OFF 0
#define BOARD_ALL_OFF_SLAVE 1
#define BOARD_ALL_ON 2
#define BOARD_ALL_ON_SLAVE 3
#define BOARD_MASTER 4

switch (board_id){
    case BOARD_ALL_OFF:
    case BOARD_ALL_OFF_SLAVE:
        all_led_off();
        Serial.println("LOCAL ALL LED OFF");
        i2c_broadcast(I2C_ALL_OFF);
        break;

     case BOARD_ALL_ON:
     case BOARD_ALL_ON_SLAVE:
        all_led_on()
        Serial.println("LOCAL ALL LED ON");
        i2c_broadcast(I2C_ALL_ON);
        break;

     case BOARD_MASTER:
     default:
        wake_set_target();
        Serial.println("LOCAL WAKE SET TARGET");
        i2c_broadcast(I2C_WAKE_SET);
        break;
```